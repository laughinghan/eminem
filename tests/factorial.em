let .factorial ^x (
  tern (x.lte 0) ^(
    1
  ) ^(
    factorial (x.minus 1) .times x
  )
)

println (factorial 0)
println (factorial 1)
println (factorial 2)
println (factorial 4)
println (factorial 8)
