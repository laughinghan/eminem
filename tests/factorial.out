1
1
2
24
40320
